src/framework/enhancement/js/ContextAwareness.js

```diff
@@ -0,0 +1,253 @@
+        var root = segs.length === 0;
+        if (root) {
+            gradeNames.push("fluid.resolveRoot");
+        }
+        var memberName = fluid.typeNameToMemberName(options.singleRootType || type);
+        segs.push(memberName);
+        fluid.construct(segs, options, instantiator);
+    };
+    
+    /** Destroy an instance created by `fluid.constructSingle`
+     * @param parentPath {String|Array of String} Parent of path where the new component is to be constructed, represented as a string or array of segments
+     * @param typeName {String} The type name used to construct the component (either `type` or `singleRootType` of the `options` argument to `fluid.constructSingle`
+     * @param instantiator {Instantiator} [optional] The instantiator holding the component to be created - if blank, the global instantiator will be used
+    */
+    fluid.destroySingle = function (parentPath, typeName, instantiator) {
+        instantiator = instantiator || fluid.globalInstantiator;
+        var segs = fluid.model.parseToSegments(parentPath, instantiator.parseEL, true);
+        var memberName = fluid.typeNameToMemberName(typeName);
+        segs.push(memberName);
+        fluid.destroy(segs, instantiator);
+    };
+    
+    fluid.defaults("fluid.contextAware.marker", {
+        gradeNames: ["fluid.component"]
+    });
+    
+    
+    // unsupported, NON-API function
+    fluid.contextAware.makeCheckMarkers = function (checks, path, instantiator) {
+        fluid.each(checks, function (value, markerTypeName) {
+            fluid.constructSingle(path, {
+                type: markerTypeName,
+                gradeNames: "fluid.contextAware.marker",
+                value: value
+            }, instantiator);
+        });
+        
+    };
+    /** Peforms the computation for `fluid.contextAware.makeChecks` and returns a hash suitable for being sent to `fluid.contextAware.makeCheckMarkers` - 
+     *
+     * @return A hash of horizon names to grade names - this can be sent to fluid.contextAware.makeCheckMarkers
+     */
+    // unsupported, NON-API function
+    fluid.contextAware.performChecks = function (checkHash) {
+        return fluid.transform(checkHash, function (checkRecord) {
+            if (fluid.isPrimitive(checkRecord)) {
+                return checkRecord;
+            } else if ("value" in checkRecord) {
+                return checkRecord.value;
+            } else if ("func" in checkRecord) {
+                return checkRecord.func();
+            } else if ("funcName" in checkRecord) {
+                return fluid.invokeGlobalFunction(checkRecord.funcName);
+            } else {
+                fluid.fail("Error in contextAwareness check record ", checkRecord, " - must contain an entry with name value, func, or funcName");
+            }
+        });
+    };
+
+    /**
+     * Takes an hash of check context names to check records, designating a collection of context markers which might be registered at a location
```

Comment:
The point is that you don't use the term consistently; have a look at other API documentation comments throughout the framework, and you'll notice the term "hash" is not widely used (and, with the exception of the <code>fluid.arrayToHash</code> function, has been largely introduced through this particular pull request).

Your point about the ambiguity of the term "object" actually supports my argument that we need to distinguish these clearly as a type of data structure used throughout the framework.

Regardless, you need to pick a term and apply it consistently throughout framework. Keep in mind the prevalence of the <code>{Object}</code> notation throughout your JSDoc-style comments. :wink: